main.tf

```diff
@@ -62,10 +62,14 @@ resource "aws_lb" "main" {
   security_groups = [aws_security_group.alb_sg.id]
   idle_timeout    = var.alb_idle_timeout
 
-  access_logs {
-    enabled = true
-    bucket  = var.logs_s3_bucket
-    prefix  = "alb/${var.name}-${var.environment}"
+  dynamic "access_logs" {
+    # Skips creating the block if logs_s3_bucket is empty string
+    for_each = var.logs_s3_bucket == "" ? [] : ["create block"]
```

Comment:
You may be running into https://github.com/hashicorp/terraform/issues/22281

So, I would move the ternary logic to locals under a new var. 
```hcl
locals {
  access_logs_buckets = var.logs_s3_bucket == "" ? [] : [var.logs_s3_bucket]
}
```
Then here, you can use that in your block converted to a map and use the key (which we just also set as the value)
```hcl
dynamic "access_logs" {
  for_each = { for v in local.access_logs_buckets : v => v }
  content {
    enabled = true
    bucket  = access_logs.key
    prefix  = "alb/${var.name}-${var.environment}"
  }
}

Reply:
Mentioned on Slack, but putting here for history: `terraform apply` works dandy. It's unclear why Terratest is losing it's marbles. For science, I've tried your suggestion. Test is running.